syntax/check.rs

```diff
@@ -233,6 +234,39 @@ fn check_type_weak_ptr(cx: &mut Check, ptr: &Ty1) {
     cx.error(ptr, "unsupported weak_ptr target type");
 }
 
+fn check_type_kj_maybe(cx: &mut Check, ptr: &Ty1) {
+    match &ptr.inner {
+        Type::Ident(ident) => {
+            if cx.types.rust.contains(&ident.rust) {
+                cx.error(ptr, "kj::Maybe of a Rust type is not supported yet");
+                return;
+            }
+
+            match Atom::from(&ident.rust) {
+                None
+                | Some(
+                    Bool | U8 | U16 | U32 | U64 | Usize | I8 | I16 | I32 | I64 | Isize | F32 | F64,
+                ) => return,
+                Some(_) => {}
+            }
+        }
+        Type::Ptr(_ptr) => {
+            return;
+            // TODO: ("Reminder to check pointer type later")
```

Comment:
Reminder to make sure these TODOs are addressed. Should we be calling `check_type_ptr()` and `check_type_ref()`? I see those two functions do exist.